example/particles/particles.cpp

```diff
@@ -192,54 +215,111 @@ TaskStatus CreateSomeParticles(MeshBlock *pmb) {
+            vz(n) = cos(theta);
+            // Project v onto plane normal to sphere
+            Real vdN = vx(n) * x(n) + vy(n) * y(n) + vz(n) * z(n);
+            Real NdN = r * r;
+            vx(n) = vx(n) - vdN / NdN * x(n);
+            vy(n) = vy(n) - vdN / NdN * y(n);
+            vz(n) = vz(n) - vdN / NdN * z(n);
+
+            // Normalize
+            Real v_tmp = sqrt(vx(n) * vx(n) + vy(n) * vy(n) + vz(n) * vz(n));
+            vx(n) *= v / v_tmp;
+            vy(n) *= v / v_tmp;
+            vz(n) *= v / v_tmp;
+
+            // Create particles at the beginning of the timestep
+            t(n) = t0;
+
+            weight(n) = 1.0;
+
+            rng_pool.free_state(rng_gen);
+          }
+        });
+  } else {
+    pmb->par_for(
+        "CreateSomeParticles", 0, swarm->GetMaxActiveIndex(), KOKKOS_LAMBDA(const int n) {
+          if (new_particles_mask(n)) {
+            auto rng_gen = rng_pool.get_state();
+
+            // Randomly sample in space in this meshblock
+            x(n) = minx_i + nx_i * dx_i * rng_gen.drand();
+            y(n) = minx_j + nx_j * dx_j * rng_gen.drand();
+            z(n) = minx_k + nx_k * dx_k * rng_gen.drand();
+
+            // Randomly sample direction on the unit sphere, fixing speed
+            Real theta = acos(2. * rng_gen.drand() - 1.);
+            Real phi = 2. * M_PI * rng_gen.drand();
+            vx(n) = v * sin(theta) * cos(phi);
+            vy(n) = v * sin(theta) * sin(phi);
+            vz(n) = v * cos(theta);
+
+            // Create particles at the beginning of the timestep
+            t(n) = t0;
+
+            weight(n) = 1.0;
+
+            rng_pool.free_state(rng_gen);
+          }
+        });
+  }
 
-          rng_pool.free_state(rng_gen);
-        }
-      });
+  swarm->swarm_num_incomplete_ = swarm->GetNumActive();
 
   return TaskStatus::complete;
 }
 
-TaskStatus TransportParticles(MeshBlock *pmb, StagedIntegrator *integrator) {
+TaskStatus TransportParticles(MeshBlock *pmb, const StagedIntegrator *integrator,
+                              const double t0) {
   auto swarm = pmb->swarm_data.Get()->Get("my particles");
+  auto pkg = pmb->packages.Get("particles_package");
+  auto orbiting_particles = pkg->Param<bool>("orbiting_particles");
 
-  int max_active_index = swarm->get_max_active_index();
+  int max_active_index = swarm->GetMaxActiveIndex();
 
   Real dt = integrator->dt;
 
-  auto &x = swarm->GetReal("x").Get();
-  auto &y = swarm->GetReal("y").Get();
-  auto &z = swarm->GetReal("z").Get();
-  const auto &vx = swarm->GetReal("vx").Get();
-  const auto &vy = swarm->GetReal("vy").Get();
-  const auto &vz = swarm->GetReal("vz").Get();
+  auto &t = swarm->Get<Real>("t").Get();
+  auto &x = swarm->Get<Real>("x").Get();
+  auto &y = swarm->Get<Real>("y").Get();
+  auto &z = swarm->Get<Real>("z").Get();
+  const auto &vx = swarm->Get<Real>("vx").Get();
+  const auto &vy = swarm->Get<Real>("vy").Get();
+  const auto &vz = swarm->Get<Real>("vz").Get();
```

Comment:
```suggestion
  auto &vx = swarm->Get<Real>("vx").Get();
  auto &vy = swarm->Get<Real>("vy").Get();
  auto &vz = swarm->Get<Real>("vz").Get();
```

In the orbiting example those are not constant anymore.

Reply:
Oops yes thanks